packages/modules/vehicles/smarteq/api.py

```diff
@@ -60,60 +83,263 @@ def nested_key_exists(element: dict, *keys: str) -> bool:
 
 class Api:
 
-    def __init__(self, vehicle: int):
+    def __init__(self, conf: SmartEQ, vehicle: int):
         self.log = logging.getLogger(__name__)
+
+        self.conf = conf
+        self.username = conf.configuration.user_id
+        self.password = conf.configuration.password
+        if self.password is None:
+            self.password = ""
+        self.pin = conf.configuration.pin
+        self.vin = conf.configuration.vin
+
+        if type(conf.configuration.useWebSocket) == bool:
+            self.useWebSocket = conf.configuration.useWebSocket
+        elif type(conf.configuration.useWebSocket) == str:
+            self.useWebSocket = json.loads(conf.configuration.useWebSocket)
+        else:
+            # if not properly defined set to False
+            self.useWebSocket = False
+
+        self.refreshToken = conf.configuration.refreshToken
+        self.refreshTokenMQTT = self.refreshToken
+
+        self.last_opmode = conf.configuration.opMode
+        if self.last_opmode is None:
+            self.last_opmode = 'unknown'
+        self.vehicle = vehicle
+        # configurable logging Filter, will come from configuration
+        self.logFilter = conf.configuration.logFilter
+        if self.logFilter is None:
+            self.logFilter = ''
+        self.write_store_count = 0
+
+        self.su = socUtils()
+
         self.storeFile = str(RAMDISK_PATH) + '/soc_smarteq_store_vh_' + str(vehicle)
+        self.ws_storeFile = str(RAMDISK_PATH) + '/soc_smarteq_store_ws'
+        self.messageFile = str(RAMDISK_PATH) + '/soc_smarteq_message_vh_' + str(vehicle)
```

Comment:
Was sind das für Dateien, die hier geschrieben werden? Die Tokens sollten im Broker, der SoC im Rückgabewert und alles andere im Log landen.

Reply:
ws_storeFile ist eine Datei, in der die Daten der websocket-Schnittstelle gespeichert werden Beispiel:
````{
    "vin": "WME4530911Y181749",
    "soc": "92",
    "range": "113",
    "ws_soc_ts": "2024-06-19 15:50:22",
    "ws_ts": 1718805022,
    "ws_connect_time": 0,
    "connect_time": "1718803134"
}````
Diese Datei hat die VIN im Namen und wird daher je VIN geschrieben.
Der "normale" SOC-Abruf kann damit ohne REST-Abruf direkt die Daten verwenden.
messageFile ist zum Test/Analyse gedacht.
Die Websocket-Schnittstelle liefert einige Message-Typen.
Die ankommenden Messages werden per typ in die Ramdisk geschrieben.
Damit lässt sich erkennen wenn sich die Daten einmal ändern sollten.